lib/src/polygon_clipping/vector_extension.dart

```diff
@@ -0,0 +1,73 @@
+import 'dart:math';
+
+import 'package:vector_math/vector_math.dart';
+
+extension Vector2Extension on Vector2 {
+  /* Given a vector, return one that is perpendicular */
+  Vector2 get perpendicularVector {
+    return Vector2(-y, x);
+  }
+}
+
+/* Get the intersection of two lines, each defined by a base point and a vector.
```

Comment:
we already have most of the vector maths implemented on `Position` type

Reply:
I'm looking at the Position; I need help finding some of the methods needed for this package. Would you like me to add them to the position class or continue using the vector package from vector math, which is supported and maintained by the Flutter and Flare teams?  

My gut reaction makes me nervous about treating Position like a point and a vector, but it might make little difference in implementation. I'll follow your lead, so please let me know what works best for turf.

Reply:
is there a difference in vector and coordinate/position in the package you're working in? Because basically, a Position is a geodetic coordinates, which is a spherical coordinate. It's not a cartesian coordinate though.

you can just do something like (maybe):
```dart
extension PositionExt on Position {
  /* Given a vector, return one that is perpendicular */
 Position get perpendicularVector {
    return Position.named(lng: -lat, lat: lng);
  }
}
```

Reply:
important to notice too is that altitude (the third dimension) in our Position is optional. So most of the times we have a 2D coordinate

Reply:
In this implementation, I don't think there's a major difference but I followed this package line by line to the best of my ability https://github.com/mfogel/polygon-clipping/tree/main

I'm well aware of the geodetic coordinate system and, but this package comments on it beyond what it says in the read me. The package distinguishes between the two events, though they only use the "x" and "y" components of both. I'm well-versed in vector math but don't see any reason we can't just use Position for both other than the naming convention unless we need or want to use the features maintained by vector_math down the line. 

In short we definitely don't need to have vectors from what I can tell right now.